### ISSUE TITLE
Crash happening at line 113 of pool_type.hpp

### ISSUE THREAD

**Issue by SGunner2014** [JGRennison/OpenTTD-patches#12]
The output is as follows:

```
Error: Assertion failed at line 113 of /home/openttd/OpenTTD-patches-jgrpp-0.17.0/src/core/pool_type.hpp: index < this->first_unused
	index: 255, first_unused: 10, name: Company
Crash encountered, generating crash log...
*** OpenTTD Crash Report ***

Crash at: Mon Feb 13 02:47:06 2017
In game date: 2025-02-01 (0, 0)

Crash reason:
 Signal:  Aborted (6)
          si_code: -6
 Message: Assertion failed at line 113 of /home/openttd/OpenTTD-patches-jgrpp-0.17.0/src/core/pool_type.hpp: index < this->first_unused
	index: 255, first_unused: 10, name: Company

Within context:
     0: DoCommandP: tile: 0x0, p1: 0x50002, p2: 0x2, company: 5 (Latias Transport), cmd: 0x150 (CmdCompanyCtrl), my_cmd: 1

OpenTTD version:
 Version:    jgrpp-0.17.0 (0)
 NewGRF ver: 17006c5f
 Bits:       64
 Endian:     little
 Dedicated:  no
 Build date: Feb 12 2017 15:03:18
 Configure:  ./configure 

Stacktrace:
 [00]           0x784039 ./openttd                                CrashLogUnix::LogStacktrace(char*, char const*) const + 0x149
 [01]           0x65305b ./openttd                                CrashLog::FillCrashLog(char*, char const*) const + 0x10b
 [02]           0x65333f ./openttd                                CrashLog::MakeCrashLog() const + 0x7f
 [03]           0x783b46 ./openttd                                + 0x383b46
 [04]     0x7f4379a7d390 /lib/x86_64-linux-gnu/libpthread.so.0    + 0x11390
 [05]     0x7f43786af428 /lib/x86_64-linux-gnu/libc.so.6          gsignal + 0x38
 [06]     0x7f43786b102a /lib/x86_64-linux-gnu/libc.so.6          abort + 0x16a
 [07]           0x76da7d ./openttd                                + 0x36da7d
 [08]           0x8f5044 ./openttd                                + 0x4f5044
 [09]           0x667e57 ./openttd                                ChangeOwnershipOfCompanyItems(Owner, Owner) + 0xc67
 [10]           0x642f11 ./openttd                                CmdCompanyCtrl(unsigned int, DoCommandFlag, unsigned int, unsigned int, char const*) + 0x161
 [11]           0x63f2ff ./openttd                                DoCommandPInternal(unsigned int, unsigned int, unsigned int, unsigned int, void (*)(CommandCost const&, unsigned int, unsigned int, unsigned int), char const*, bool, bool, unsigned int) + 0x2af
 [12]           0x63ffd2 ./openttd                                DoCommandP(unsigned int, unsigned int, unsigned int, unsigned int, void (*)(CommandCost const&, unsigned int, unsigned int, unsigned int), char const*, bool, unsigned int) + 0x162
 [13]           0x640293 ./openttd                                DoCommandP(CommandContainer const*, bool) + 0x23
 [14]           0x70c8e6 ./openttd                                NetworkExecuteLocalCommandQueue() + 0x86
 [15]           0x70491c ./openttd                                NetworkGameLoop() + 0xdc
 [16]           0x7716b3 ./openttd                                GameLoop() + 0xa3
 [17]           0x913661 ./openttd                                VideoDriver_Dedicated::MainLoop() + 0x291
 [18]           0x76f22f ./openttd                                openttd_main(int, char**) + 0x13af
 [19]     0x7f437869a830 /lib/x86_64-linux-gnu/libc.so.6          __libc_start_main + 0xf0
 [20]           0x59bd99 ./openttd                                _start + 0x29

Registers:
 rax:                0 rbx:   0x7ffd6cfba10f rcx:   0x7f43786af428 rdx:              0x6
 rsi:           0x7ef7 rdi:           0x7ef7 rbp:         0x941d00 rsp:   0x7ffd6cfb97b8
 r8:    0x7f437a18b8c0 r9:              0xb2 r10:              0x8 r11:            0x202
 r12:         0x148368 r13:          0x520da r14:             0xff r15:             0xa6
 rip:   0x7f43786af428 eflags:    0x202

Operating system:
 Name:     Linux
 Release:  4.4.0-62-generic
 Version:  #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017
 Machine:  x86_64
 Compiler: GCC 5.4.0 "5.4.0 20160609"

Distro version:
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.1 LTS
Release:	16.04
Codename:	xenial

Configuration:
 Blitter:      null
 Graphics set: OpenGFX (5580)
 Language:     /home/openttd/OpenTTD-patches-jgrpp-0.17.0/bin/lang/english_US.lng
 Music driver: null
 Music set:    NoMusic (0)
 Network:      server
 Sound driver: null
 Sound set:    NoSound (2)
 Video driver: dedicated

Fonts:
 Small:  sprite
 Medium: sprite
 Large:  sprite
 Mono:   sprite

AI Configuration (local: 255) (current: 5):
  0: Human
  1: Human
  2: Human
  3: Human
  4: Human
  5: Human
  6: Human
  7: Human
  8: Human
  9: Human

Libraries:
 LZMA:       5.1.0alpha
 LZO:        2.08
 PNG:        1.2.54
 SDL:        1.2.15
 Zlib:       1.2.8

---- gamelog start ----
Tick 0: new game started
Revision text changed to jgrpp-0.17.0, savegame version 195, not modified, _openttd_newgrf_version = 0x17006c5f
New game mode: 2 landscape: 0
Added NewGRF: GRF ID 4D42000A, checksum BC3A0E791C74C1A04A3B90C3B2FA83A6, filename: bats-0.4/batsv0.4w.grf (md5sum matches)
Added NewGRF: GRF ID 4D420100, checksum 1A5D28D0719FCDB928F59AEFDBE07B7C, filename: bobs_british_buses-0.2.1/bbbw.grf (md5sum matches)
Added NewGRF: GRF ID 4A430001, checksum 4CF50E2A770F6672A77A8842A3129ECA, filename: british_stations_set-0.0.4final/dstatsw.grf (md5sum matches)
Added NewGRF: GRF ID 42525300, checksum 85CF3A21072391B6877803CC8D3A12AB, filename: bros__british_rail_openttd_set-r52/bros.grf (md5sum matches)
Added NewGRF: GRF ID 54530102, checksum 0ACA956A9A97F341D27173B69C3F203A, filename: cfl_trains_luxembourgish_trains-1.04/cflsettrains.grf (md5sum matches)
Added NewGRF: GRF ID 43485053, checksum 3C4699AA326D6397DA937AD436DE8906, filename: chips_station_set-1.7.0/chips.grf (md5sum matches)
Added NewGRF: GRF ID 52455400, checksum 1600687C50C5AB1E72EDD487CAD50E2A, filename: coded_by_aegir._artwork_by_aegir-1/ae_cityw.grf (md5sum matches)
Added NewGRF: GRF ID FE16215C, checksum 69D13699AC16604E825F7657E6AACA72, filename: city_tram_set-b02/cts02.grf (md5sum matches)
Added NewGRF: GRF ID 444C1105, checksum 4AC44465E334F062F56BC24C728C8E72, filename: dach_trains-0.3.2/dach.grf (md5sum matches)
Added NewGRF: GRF ID FBFB0502, checksum 55AE61673CBCDE4A7231DEA85973278C, filename: dutch_road_furn._uk_signs_addon-1.0.0/dutchroadfurnitureuksignsaddon.grf (md5sum matches)
Added NewGRF: GRF ID FBFB0501, checksum 60082E7DF1D3976274AB86843C3E07B2, filename: dutch_road_furniture-0.7.1/dutchroadfurniture.grf (md5sum matches)
Added NewGRF: GRF ID 58534453, checksum 2E100BCA91DB8371CD9D57DBEC3AB0C8, filename: dutch_stations_set-1.0.0alpha/dutchstatw.grf (md5sum matches)
Added NewGRF: GRF ID 504A0111, checksum A118EB57401A4486F9B00FCFFDDA689C, filename: dutch_stations_additionset_part2-0.1/dstatadd_part2_v0_1.grf (md5sum matches)
Added NewGRF: GRF ID 504A0112, checksum B5A3C1AF5C32C0430C374C66963C3320, filename: dutch_station_addition_part_2-0.3/dstatadd_part2_v0_3.grf (md5sum matches)
Added NewGRF: GRF ID 504A0110, checksum E8244C5278CB2408A818854DC11F8940, filename: dutch_station_addition_set-0.8/dstatadd_v0_8.grf (md5sum matches)
Added NewGRF: GRF ID FBFB0601, checksum E167E84751FD2E61967944010AF5C56D, filename: dutch_trainset_2-2.1.0/dutchtrains.grf (md5sum matches)
Added NewGRF: GRF ID FBFB9906, checksum 0D9BEFCA384AAE118DACEF9FEF044D69, filename: dutch_trainset_cargo_extension-1/dutchtrainscargoext.grf (md5sum matches)
Added NewGRF: GRF ID FBFB0102, checksum CE6B0C978C025F3477CC9A91AE9109FE, filename: dutch_tram_set-2.0.2/dutchtramset.grf (md5sum matches)
Added NewGRF: GRF ID 44442303, checksum 90300B85C4DE1DD192691827B054522D, filename: uk_railway_set_tracks-2.0/pb_trax.grf (md5sum matches)
Added NewGRF: GRF ID 414E0101, checksum 5D665C64693A438E3046C22666F46E08, filename: fish_ship_set-0.9.2/fish.grf (md5sum matches)
Added NewGRF: GRF ID FBFB0301, checksum D961A46B2CC37A04CD4FCBEDD2311A53, filename: foobars_tram_tracks-1.0.1/foobarstramtracks.grf (md5sum matches)
Added NewGRF: GRF ID FBFB0901, checksum 4619642A3BF1A535AEDBDA426FF63EBB, filename: friss-0.5.2/friss-stations.grf (md5sum matches)
Added NewGRF: GRF ID 52570103, checksum EDB2D26F9D81423734CA12412F309CF3, filename: generic_tram_set.0.4/generictrams.grf (md5sum matches)
Added NewGRF: GRF ID 4A430002, checksum 93EAC5F5396584B92D4C5F0AAAF3F6C5, filename: industrial_stations_renewal-1.0.2/indstatr.grf (md5sum matches)
Added NewGRF: GRF ID 524B0301, checksum BB7BBE4FF23B349D17ED0A682503680C, filename: lightrailtramtracks-0.2/tramtrkw.grf (md5sum matches)
Added NewGRF: GRF ID 4D656F17, checksum 8A4F8A04256C90BECF818D134C91C816, filename: long_vehicles-v404oct2008/4lvw.grf (md5sum matches)
Added NewGRF: GRF ID FBFB9905, checksum F747797A14AAA646402DC4E16DE5E82C, filename: longer_girder__steel_bridge.1/longergirdersteelbridge.grf (md5sum matches)
Added NewGRF: GRF ID 4642FB21, checksum 117F6F1F07C18A9C15ADC6127AFBAE97, filename: modern_tram_set-3.1/motraset_r3.1.grf (md5sum matches)
Added NewGRF: GRF ID 686D6301, checksum ECFF8464F08FD6772E7A133C46FA43F2, filename: more_british_town_names-1.0/morebritishtownnames_1.grf (md5sum matches)
Added NewGRF: GRF ID 4A44FAB1, checksum 81D015C3929F1EC2C34574BE7D6420E0, filename: newcc_set-0.2/newcc.grf (md5sum matches)
Added NewGRF: GRF ID 576D0301, checksum 69D589CBE85C11DA52BC31E7729D7224, filename: progressive_rail_set-2/progressive_rails.2.0.2.grf (md5sum matches)
Added NewGRF: GRF ID 52455200, checksum 0E5A9C5ACD2BCAF7FA0658C3F905A660, filename: rural_stations-1/ae_ruraw.grf (md5sum matches)
Added NewGRF: GRF ID 44508550, checksum BEE9D0982856E627DDF24902019413F8, filename: sbb_set-1.0e/sbbset.grf (md5sum matches)
Added NewGRF: GRF ID 52455300, checksum 37C0B839B13C4B18051580C81EEDF427, filename: suburban_stations-1/ae_subuw.grf (md5sum matches)
Added NewGRF: GRF ID 74677462, checksum 1D2A4E03D9F30922788C166254C27619, filename: total_bridge_renewal_set-1.2/total_bridges.grf (md5sum matches)
Added NewGRF: GRF ID 44442411, checksum E6A38D0DC67DE54DD2928447BB72F924, filename: tai_uk_houses-gamma1/tai_uk.grf (md5sum matches)
Added NewGRF: GRF ID 54560301, checksum ED22416EE087725E4B5A9AF8F0357F29, filename: trolleybus_set-0.2/trolley.grf (md5sum matches)
Added NewGRF: GRF ID 44441000, checksum DA36131A815D2D5FC158C7A27D67B5AF, filename: ukrs2__the_uk_railway_set-1.05/proj1000.grf (md5sum matches)
Added NewGRF: GRF ID 44441001, checksum 83915879F68DD174C953868C0629A4C9, filename: ukrs2_uk_railway_addon_set-gobi/pj1kadd.grf (md5sum matches)
Added NewGRF: GRF ID 44442202, checksum 3DAC7433C0DA14AD3C5E8C89D73A6A2E, filename: uk_renewal_set_addon_pack.3.04/ukrsap1w.grf (md5sum matches)
Added NewGRF: GRF ID 44440111, checksum 9B5051E70D60EF2926A249B923AFB0CA, filename: uk_renewal_set.3.04/pb_ukrs.grf (md5sum matches)
Added NewGRF: GRF ID 72777473, checksum EBEAB33B398DE7F2921F853C496F6F91, filename: uk_town_set-2.2/ukts_2.2.grf (md5sum matches)
Added NewGRF: GRF ID 504E0001, checksum E7781323D2B0086E2E76342DAA036054, filename: uk_waypoints-1/ukwaypointsw.grf (md5sum matches)
Added NewGRF: GRF ID 444E0400, checksum 735D3D74D5DCADBF1FD6EF7A9C5A3519, filename: vast_station_tiles-0.2.0/vast.grf (md5sum matches)
Added NewGRF: GRF ID 57A50001, checksum 318F3C3F17CA68EAD533BAC88B093094, filename: was_world_airliners_set-0.1/was.grf (md5sum matches)
Added NewGRF: GRF ID F1250005, checksum 04817A8E04171941B350B8363AF798AB, filename: firs_industry_replacement_set-1.4.5/firs.grf (md5sum matches)
Tick 1280: game loaded
New game mode: 1 landscape: 0
Tick 1369: settings changed
Setting changed: construction.command_pause_level : 1 -> 3
---- gamelog end ----

*** End of OpenTTD Crash Report ***

Crash log generated.

Writing crash log to disk...
Crash log written to /root/.openttd/crash.log. Please add this file to any bug reports.

Writing crash savegame...
Crash savegame written to /root/.openttd/crash.sav. Please add this file and the last (auto)save to any bug reports.

Writing crash screenshot...
Writing crash screenshot failed.
```

**Comment by JGRennison** [comment]
This bug should be already fixed in e1ee952b.
I'll do another release soon which includes this.
Many thanks for reporting the issue.

**Comment by JGRennison** [comment]
This should be fixed in v0.17.1 now.
